src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -703,6 +737,9 @@ function VisualizeAppController(
     searchSource.setField('index', currentIndex);
     searchSource.setParent(searchSourceGrandparent);
 
+    delete savedVis.savedSearchId;
+    delete vis.data.savedSearchId;
```

Comment:
We are handling the `linked` state in url. Callinng `stateContainer.transitions.unlinkSavedSearch` will set `linked: false`, which will remove `savedSearchId` from both `savedVis` and `vis.data` objects in `handleLinkedSearch`, so there is no necessity to do it here.